examples/sneakyinvoke.jl

```diff
@@ -0,0 +1,23 @@
+using IRTools: @dynamo, argument!, IR
+
+
+# Implementation of `invoke(f, argtypes::Type, args...`, where `args` don't have
+# to conform to `argtypes`.  I.e., it allows you to call `f` on `args`, but using the
+# method for `argtypes`.
+# See https://julialang.slack.com/archives/CJ357CY2Y/p1579598240011500.
+
+concrete(::Type{Type{T}}) where {T} = T
+
+function _sneaky_transform(f::Type, Types::Type{<:Tuple})
```

Comment:
Is it necessary to split this out? I can see this being useful for development but for show it off we should probably make it as simple as possible.

Reply:
Not at all. But accessing `parameters[1]` just felt kind of ugly to me.

Reply:
For that you need `concrete` but I don't see that it needs `_sneaky_transform` to be split out, because that's not doing any unwrapping, it just passes through arguments without modification.

It could work well to just do `@dynamo function sneakyinvoke(f, Type{T}, ...) where T<:Tuple`. The dynamo type signature refers to the unwrapped parameters so that might be a cleaner way to unwrap that avoids the utility functions.

Reply:
I was wrong above due to #41, but I've pushed a fix which makes the following work:

```julia
@dynamo function sneakyinvoke(f, ::Type{T}, args...) where T<:Tuple
    ir = IR(f, T.parameters...)
    argument!(ir, at = 2)
    return ir
end
```

I think this is a slightly more idiomatic way to do the unwrwapping.